class.jetpack.php

```diff
@@ -4791,6 +4791,14 @@ function build_connect_url( $raw = false, $redirect = false, $from = false, $reg
 			$url = add_query_arg( 'from', $from, $url );
 		}
 
+		// Ensure that class to get the affiliate code is loaded
+		if ( ! class_exists( 'Jetpack_Affiliate' ) ) {
+			require_once JETPACK__PLUGIN_DIR . 'class.jetpack-affiliate.php';
+		}
+		// Get affiliate code and add it to the URL
+		if ( '' !== $aff = Jetpack_Affiliate::init()->get_affiliate_code() ) {
```

Comment:
I'd suggest adding some `()` around the assignment here. Maybe something like:

```php
if ( '' !== ( $aff = Jetpack_Affiliate::init()->get_affiliate_code() ) )
```

Reply:
👍